diopter/bisector.py

```diff
@@ -1,258 +1,165 @@
-import logging
-import math
 from abc import ABC, abstractmethod
 from pathlib import Path
+from subprocess import CalledProcessError
+from tempfile import TemporaryDirectory
+from typing import TypeAlias
 
-from ccbuilder import Commit, CompilerProject, Repo, Revision, find_cached_revisions
-
-
-class BisectionException(Exception):
-    pass
-
-
-def find_sorted_cached_commits_from_range(
-    good_rev: Revision,
-    bad_rev: Revision,
-    project: CompilerProject,
-    repo: Repo,
-    cache_prefix: Path,
-) -> list[Commit]:
-    """Return all cached commits that are in the range R=[good_rev, bad_rev] such that
-    R[i+1] is ancestor of R[i].
-
-    Args:
-        good_rev (Revision): good_rev
-        bad_rev (Revision): bad_rev
-        project (CompilerProject): project
-        repo (Repo): Repository for `project`
-        cache_prefix (Path): Path to the cache.
-
-    Returns:
-        list[Commit]:
-            All cached commits in [good_rev, bad_rev] ordered via the ancestor relation.
-    """
-    possible_revs = repo.direct_first_parent_path(good_rev, bad_rev)
-
-    cached_revs = find_cached_revisions(project, cache_prefix)
-    cached_revs = [r for r in cached_revs if r in possible_revs]
-
-    # Create enumeration dict to sort cached_revs with
-    sort_dict = dict((r, v) for v, r in enumerate(possible_revs))
-    cached_revs = sorted(cached_revs, key=lambda x: sort_dict[x])
-    return cached_revs
-
-
-def _get_midpoint_after_failure(
-    bad_rev: Revision,
-    good_rev: Revision,
-    midpoint: Revision,
-    failed_to_build_counter: int,
-    repo: Repo,
-    max_failed_builds: int = 3,
-) -> Commit:
-    if failed_to_build_counter >= max_failed_builds:
-        raise BisectionException(
-            "Failed too many times in a row while bisecting. Aborting bisection..."
-        )
-    if failed_to_build_counter % 2 == 0:
-        # Get size of range
-        range_size = len(repo.direct_first_parent_path(midpoint, bad_rev))
-
-        # Move 10% towards the last bad
-        step = max(int(0.9 * range_size), 1)
-        midpoint = repo.rev_to_commit(f"{bad_rev}~{step}")
+from diopter.utils import run_cmd
+
+GitRevision: TypeAlias = str
+
```

Comment:
Do you want to differentiate between a revision and a commit hash?
The commit hash is 'pure' and (mostly) won't change while a revision (`main`) potentially can.

Reply:
That was stale. I forgot to delete it

Reply:
Yeah that's something I never did and I cost me a confusing debug session once.